src/LuaPrinter.ts

```diff
@@ -59,22 +59,23 @@ export class LuaPrinter {
 
         const rootSourceNode = this.printImplementation(block, luaLibFeatures, sourceFile);
 
-        const codeWithSourceMap = rootSourceNode
-            // TODO is the file: part really required? and should this be handled in the printer?
-            .toStringWithSourceMap({file: path.basename(sourceFile, path.extname(sourceFile)) + ".lua"});
+        const sourceRoot = this.options.sourceRoot
+            || (this.options.outDir ? path.relative(this.options.outDir, this.options.rootDir || process.cwd()) : ".");
```

Comment:
Tsc makes the source filename either only the filename or maybe a relative path (Definitely no absolute paths), but this does produce absolute paths, which should probably be changed.